src/Catalog/Dnx/DnxMaker.cs

```diff
@@ -195,7 +241,27 @@ private StorageContent CreateContent(IEnumerable<string> versions)
         private async Task<Uri> SaveNupkgAsync(Stream nupkgStream, Storage storage, string id, string version, CancellationToken cancellationToken)
         {
             Uri nupkgUri = new Uri(storage.BaseAddress, GetRelativeAddressNupkg(id, version));
-            await storage.Save(nupkgUri, new StreamStorageContent(nupkgStream, "application/octet-stream", "max-age=120"), cancellationToken);
+            var content = new StreamStorageContent(
+                nupkgStream,
+                DnxConstants.ApplicationOctetStreamContentType,
+                DnxConstants.DefaultCacheControl);
+
+            await storage.Save(nupkgUri, content, cancellationToken);
+
+            return nupkgUri;
+        }
+
+        private async Task<Uri> CopyNupkgAsync(CloudBlockBlob sourceBlob, Storage storage, string id, string version, CancellationToken cancellationToken)
+        {
+            var relativeUri = GetRelativeAddressNupkg(id, version);
+            var nupkgUri = new Uri(storage.BaseAddress, relativeUri);
+            var content = new AzureServerSideCopiableContent(
+                sourceBlob,
+                DnxConstants.ApplicationOctetStreamContentType,
+                DnxConstants.DefaultCacheControl);
+
+            await storage.Save(nupkgUri, content, cancellationToken);
```

Comment:
Why is this `Save`? The implementation must check types to make this work. Seems like a broken abstraction. Can't we have a `Copy` method that only knows how to copy? And let `Save` only upload?